R/undesirable_function_linter.R

```diff
@@ -1,12 +1,50 @@
-#' @describeIn linters  Report the use of undesirable functions, e.g. \code{return}, \code{options},
-#'                      or \code{sapply} and suggest an alternative.
-#' @param fun  Named character vector, where the names are the names of the undesirable functions,
-#'             and the values are the text for the alternative function to use (or \code{NA}).
+#' @describeIn linters Report the use of undesirable functions, e.g.
+#'   \code{return}, \code{options}, or \code{sapply} and suggest an alternative.
+#' @param fun Named character vector, where the names are the names of the
+#'   undesirable functions, and the values are the text for the alternative
+#'   function to use (or \code{NA}).
+#' @param symbol_is_undesirable Whether to consider the use of an undesirable
+#'   function name as a symbol undesirable or not.
+#'
+#' @examples
```

Comment:
I realize there are no other examples and that this probably can't remain here, because it would be weird. But I had a hard time getting started with linter customization, due to the lack of examples.

Is there somewhere content like this could go?

Sidebar: It might be useful to have a function for example-ing that makes it easy to lint a character vector like this.

The `unclass()` calls were to make the example easier to write while inside RStudio, i.e. I actually wanted plain output. Presumably there is a better way to go about this and I just didn't discover it.

I know this section probably be deleted, but I wanted to pose these questions.

Reply:
>  But I had a hard time getting started with linter customization, due to the lack of examples.

Agreed! It's part of what I had in mind when documenting `get_source_expressions` more extensively:

https://github.com/jimhester/lintr/pull/562/files

Does that go in the direction you had in mind? If so it would be great if we could build on that.

Reply:
I meant something simpler: I had a hard time figuring out how to use the undesirable function linter, how to use it with non-default undesirable functions, etc. I think I had to read the test and search GitHub to try to find actual usage?

With almost 40 linters documented together, it becomes almost impossible to give much info or examples on how to use an individual linter.

I know this problem is not going to get worked out in the PR and I don't want it to become a blocker!

Reply:
I now understand how to lint inlined source. So I've made my examples cleaner, but I still understand they probably have to be deleted.

The printing problem is covered by #590.

Reply:
We could probably get examples into the docs once we have a clear plan for #510.

Reply:
I've deleted the examples.